docs/current/guides/snippets/use-services/environment-variables/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"dagger.io/dagger"
+)
+
+func main() {
+	// create dagger client
+	ctx := context.Background()
+	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		panic(err)
+	}
+	defer client.Close()
+
+	// Container setup
+	ctr := client.Container().From("alpine")
+
+	// Environment Variables
+	envs := map[string]string{
+		"ENV_VAR_1": "VALUE 1",
+		"ENV_VAR_2": "VALUE 2",
+		"ENV_VAR_3": "VALUE 3",
+	}
+
+	ctr = ctr.With(envVariables(envs))
+
+	// Execute the environment command to print env variables
+	out, err := ctr.WithExec([]string{"env"}).Stdout(ctx)
```

Comment:
The point of `.With` is to not break the chain. You've done that in Typescript, do the same in the others. 🙂